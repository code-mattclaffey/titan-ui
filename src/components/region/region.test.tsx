import React from 'react';
import { render, cleanup } from '@testing-library/react';
import { Region, RegionInner } from '.';

afterEach(() => {
    cleanup();
});

describe('<Region />', () => {
    it('should match snapshot', () => {
        const { container } = render(
            <Region>
                <RegionInner>
                    <p>Some text</p>
                </RegionInner>
            </Region>,
        );
        expect(container).toMatchSnapshot();
    });

    it('should have a thin class', () => {
        const { queryByTestId } = render(
            <Region>
                <RegionInner thin>
                    <p>Some text</p>
                </RegionInner>
            </Region>,
        );
        expect(
            queryByTestId('region-inner').classList.contains(
                'o-region__inner--thin',
            ),
        ).toEqual(true);
    });
});
