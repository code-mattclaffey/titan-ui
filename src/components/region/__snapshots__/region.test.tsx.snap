// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Region /> should match snapshot 1`] = `
<div>
  <section
    class="o-region"
  >
    <div
      class="o-region__inner"
      data-testid="region-inner"
    />
  </section>
</div>
`;
